RadixWallet/Core/SharedModels/Assets/ResourceAmount.swift

```diff
@@ -0,0 +1,52 @@
+// MARK: - ResourceAmount
+public struct ResourceAmount: Sendable, Hashable, Codable {
+	let nominalAmount: RETDecimal
+	var fiatWorth: FiatWorth?
+
+	enum CodingKeys: CodingKey {
+		case nominalAmount
+	}
+
+	init(nominalAmount: RETDecimal, fiatWorth: FiatWorth? = nil) {
+		self.nominalAmount = nominalAmount
+		self.fiatWorth = fiatWorth
+	}
+
+	public init(from decoder: Decoder) throws {
+		let container = try decoder.container(keyedBy: CodingKeys.self)
+		self.nominalAmount = try container.decode(RETDecimal.self, forKey: .nominalAmount)
+		self.fiatWorth = nil
+	}
+
+	public func encode(to encoder: Encoder) throws {
+		var container = encoder.container(keyedBy: CodingKeys.self)
```

Comment:
Note, fiatWorth is not encoded/decoded, it is always populated in AccountPortfolio when resources are fetched.